internal/resources/providers/aws_cis/monitoring/filter_pattern_parser.go

```diff
@@ -0,0 +1,275 @@
+			exp, closingParenthesisPos, err := resolveParenthesis(s, depth, i)
+			if err != nil {
+				return MetricFilterPattern{}, err
+			}
+			expressions = append(expressions, exp)
+			pointer = closingParenthesisPos + i + 1 // move pointer to the end of what has been already processed
+			continue
+		}
+
+		if _, err := buf.WriteRune(r); err != nil {
+			return MetricFilterPattern{}, fmt.Errorf("could not write rune %v", err)
+		}
+
+		tmpString := buf.String()
+		if contains, op := hasSuffixLogicalOp(tmpString); contains { // && or || marks the end of a Simple MetricFilterPattern
+			if logicalOp == "" { // set the Operator of the MetricFilterPattern without overriding it
+				logicalOp = op
+			}
+
+			if logicalOp != op {
+				return MetricFilterPattern{}, errors.New("not supported comparison with alternating logical operators")
+			}
+
+			cleanSimpleExpression := strings.TrimSuffix(tmpString, string(op))
+
+			var err error
+			expressions, err = appendSimpleExpression(cleanSimpleExpression, expressions)
+			if err != nil {
+				return MetricFilterPattern{}, err
+			}
+
+			buf.Reset()
+			buf.Grow(len(s) - i)
+		}
+	}
+
+	expressions, err := appendSimpleExpression(buf.String(), expressions)
+	if err != nil {
+		return MetricFilterPattern{}, err
+	}
+
+	if len(expressions) == 1 { // unwrap Simple Expressions
+		return expressions[0], nil
+	}
+
+	return newComplexExpression(logicalOp, expressions...), nil
+}
+
+func appendSimpleExpression(s string, expressions []MetricFilterPattern) ([]MetricFilterPattern, error) {
+	expStr := strings.TrimSpace(s)
+	// if the length is zero it means we had an already processed Complex Expressions (between parenthesis)
+	if len(expStr) > 0 {
+		exp, err := parseSimpleStatement(expStr)
+		if err != nil {
+			return nil, err
+		}
+
+		expressions = append(expressions, exp)
+	}
+	return expressions, nil
+}
+
+func resolveParenthesis(s string, depth int, i int) (MetricFilterPattern, int, error) {
+	closingParenthesisPos := matchingParenthesisPos(s[i:])
+	if closingParenthesisPos < 0 {
+		return MetricFilterPattern{}, -1, errors.New("broken parenthesis")
+	}
+
+	subS := s[i+1 : closingParenthesisPos+i] // take what is inside matching parenthesis
+	exp, err := safeParse(subS, depth+1)     // safe sub MetricFilterPattern
+	if err != nil {
+		return MetricFilterPattern{}, -1, err
+	}
+	return exp, closingParenthesisPos, nil
+}
+
+func matchingParenthesisPos(s string) int {
+	parenthesisStack := 0
+	for i, r := range s {
+		if r == '(' {
+			parenthesisStack++
+		}
+
+		if r == ')' {
+			parenthesisStack--
+		}
+
+		if parenthesisStack == 0 {
+			return i
+		}
+	}
+
+	return -1
+}
+
+func parseSimpleStatement(s string) (MetricFilterPattern, error) {
+	buf := strings.Builder{}
+	buf.Grow(len(s))
+
+	var left string
+	var operator comparisonOperator
+	foundOp := false
+
+	for i, r := range s {
+		if buf.Len() == 0 && (r == ' ' || r == '(') { // ignore trailing spaces and (
```

Comment:
wouldn't an opening parentheses would be unexpected here? I would expect handling it would have been done at a higher level (e.g., in safeParse or resolveParenthesis).